custom_components/frigate/media_source.py

```diff
@@ -666,12 +682,17 @@ async def async_browse_media(
                     )
             return base
 
+        instance_id = self._get_default_frigate_instance_id()
         identifier = Identifier.from_str(
             item.identifier,
-            default_frigate_instance_id=self._get_default_frigate_instance_id(),
+            default_frigate_instance_id=instance_id,
         )
 
-        if isinstance(identifier, EventSearchIdentifier):
+        if (
+            instance_id
+            and self._is_allowed_as_media_source(instance_id)
```

Comment:
This doesn't look right, as it'll always check the default instance_id, rather than the instance_id that may have been included in the identifier. Why can't you do what you did elsewhere and just call `self._is_allowed_as_media_source(identifier.frigate_instance_id)` ?

Reply:
It was complaining that that wasn't a valid field. It only has the Recording, Event, etc values

Reply:
This is what it is specifically:

`Item "None" of "Union[EventSearchIdentifier, EventIdentifier, RecordingIdentifier, None]" has no attribute "frigate_instance_id"`

Reply:
Is that from `mypy` on precommit? I think that's just telling you the value could be none. As long as you verify it's not none in the if statement it should work, e.g.:

`if identifier is not None and self._is_allowed_as_media_source(identifier.frigate_instance_id)` 

... may work?